gax-grpc/src/main/java/com/google/api/gax/grpc/ApiResultRetryAlgorithm.java

```diff
@@ -42,7 +42,7 @@
   public TimedAttemptSettings createNextAttempt(
       Throwable prevThrowable, ResponseT prevResponse, TimedAttemptSettings prevSettings) {
     if (prevThrowable != null
-        && ((ApiException) prevThrowable).getStatusCode() == Code.DEADLINE_EXCEEDED) {
+        && ((GrpcApiException) prevThrowable).getStatusCode().getCode() == Code.DEADLINE_EXCEEDED) {
```

Comment:
Yes, there is a `Preconditions.checkNotNull` check in the parent class (`ApiException`).